### PR TITLE
Remove nonsensical CLinearMachine::compute_bias

### PR DESCRIPTION
**Pull request by karlnapf**
* add bias coputation to CLinearRidgeRegression
* activate bias in LibLinear and LibLinearRegression by default
* cleanups

### CODE REVIEW

File: src/shogun/features/DenseFeatures.cpp [hunk 1]
```diff
@@ -88,6 +88,10 @@ template<class ST> CFeatures* CDenseFeatures<ST>::duplicate() const
 	return new CDenseFeatures<ST>(*this);
 }
 
+{
+	return result;
+}
```
Comment (iglesias) [inline src/shogun/features/DenseFeatures.cpp]: This must have slipped though.

File: src/shogun/labels/RegressionLabels.cpp [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #include <shogun/labels/DenseLabels.h>
 #include <shogun/labels/RegressionLabels.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
```
Comment (iglesias) [inline src/shogun/labels/RegressionLabels.cpp]: Not necessary or was implicit include?

File: src/shogun/regression/LinearRidgeRegression.h [hunk 3]
```diff
@@ -19,21 +19,18 @@ namespace shogun
 /** @brief Class LinearRidgeRegression implements Ridge Regression - a regularized least square
  * method for classification and regression.
  *
- * RR is closely related to Fishers Linear Discriminant (cf. LDA).
- *
  * Internally, it is solved via minimizing the following system
  *
  * \f[
- * \frac{1}{2}\left(\sum_{i=1}^N(y_i-{\bf w}\cdot {\bf x}_i)^2 + \tau||{\bf w}||^2\right)
+ * \frac{1}{2}\left(\sum_{i=1}^N(y_i-({\bf w}\cdot {\bf x}_i+b))^2 + \tau||{\bf w}||^2\right)
```
Comment (iglesias) [inline src/shogun/regression/LinearRidgeRegression.h]: Cost for the bias term as well? E.g. inside the l2-squared.

File: src/shogun/regression/LinearRidgeRegression.h [hunk 4]
```diff
@@ -19,21 +19,18 @@ namespace shogun
...
  * \f]
  *
  * which boils down to solving a linear system
  *
  * \f[
  * {\bf w} = \left(\tau {\bf I}+ \sum_{i=1}^N{\bf x}_i{\bf x}_i^T\right)^{-1}\left(\sum_{i=1}^N y_i{\bf x}_i\right)
+ * b = \frac{1}{N}/sum_{i=1}^Ny_i-{\bf w}\cdot /frac{1}{N}/sum_{i=1}^N{\bf x}_i
```
Comment (iglesias) [inline src/shogun/regression/LinearRidgeRegression.h]: Oh, I had thought without the inner/second sum.

File: src/shogun/features/DenseFeatures.cpp (hunk 1 above)
Comment (karlnapf) [reply src/shogun/features/DenseFeatures.cpp]: y

File: src/shogun/labels/RegressionLabels.cpp (hunk 2 above)
Comment (karlnapf) [reply src/shogun/labels/RegressionLabels.cpp]: unnecessary

File: src/shogun/regression/LinearRidgeRegression.cpp [hunk 5]
```diff
@@ -71,11 +71,14 @@ bool CLinearRidgeRegression::train_machine(CFeatures* data)
 	linalg::matrix_prod(feats_matrix, feats_matrix, kernel_matrix, false, true);
 	linalg::add_diag(kernel_matrix, tau_vector);
 
-	auto labels = ((CRegressionLabels*)m_labels)->get_labels();
```
[the remark this answers is not in the document]
Comment (karlnapf) [reply src/shogun/regression/LinearRidgeRegression.cpp]: will do autoformat once this works, thx

**Comment by karlnapf** [comment]
@iglesias actually the bias in the previous version was still wrong, I double checked things. Now it should work.
I will also add a way to compute this for D>N for which the current method will fail (covariance matrix doesnt have full rank anymore)

**Comment by karlnapf** [comment]
Travis times out, but all tests pass locally, merging

**Comment by iglesias** [comment]
@karlnapf all right. I did not check the implementation of the equations 8-D

**Pull request merged**